integration/spark/src/main/scala/org/apache/spark/sql/events/MergeIndexEventListener.scala

```diff
@@ -140,19 +141,47 @@ class MergeIndexEventListener extends OperationEventListener with Logging {
                   .get
                   .filterNot(streamingSegment.contains(_))
               }
+            validSegments.foreach { segment =>
+              if (segmentsToMerge.contains(segment.getSegmentNo)) {
```

Comment:
move this if check as `.filter` before ` foreach`

Reply:
Done